### DISCUSSION TITLE
redis authentication failing

### DISCUSSION THREAD

**Discussion by Aravindms369** [emqx/emqx#8196] [Q&A]
emqx-version:4.3.10
redis - Azure redis cache

we are getting below logs while client trying to connect to redis through emqx_auth_redis plugin for authentication:

[warning] ****@**** [Channel] Client **** (Username: '****') login failed for not_authorized
[error] ****@**** [Redis] Auth from redis failed: not_authorized

Please help in the same to resolve.

**Comment by zmstone** [comment]
First question come to my mind: Is redis in azure anything different in client authentication?
Then, is this config set?

https://github.com/emqx/emqx/blob/4b867e6536695a04030215671dde06d8ec1765ea/apps/emqx_auth_redis/etc/emqx_auth_redis.conf#L37

**Reply by Aravindms369** [reply]
Hi @zmstone ,

Usually we specify the Redis server and its password in this configuration file to which the client can access for validating its credentials. We use the azure redis cache server for this purpose. In this file we specify the azure redis cache credentials.
